src/sagemaker/workflow/pipeline.py

```diff
@@ -59,9 +62,15 @@ class Pipeline(Entity):
     parameters: Sequence[Parameter] = attr.ib(factory=list)
     steps: Sequence[Union[Step, StepCollection]] = attr.ib(factory=list)
     sagemaker_session: Session = attr.ib(factory=Session)
+    repo = Repo("~/.git")
```

Comment:
You can't assume the SDK is in a git repo. This will throw `git.exc.NoSuchPathError` if it isn't.

Reply:
Also this is reading from the user's home directory, not the directory root